### PR TITLE
Add documentation for kinfo_proc fields

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
It makes it much easier considering the huge number of fields...

### CODE REVIEW

**Comment by JohnTitor** [comment]
I've moved the declaration in #2545 so this will need a rebase after it's merged.

**Comment by GuillaumeGomez** [comment]
Ah interesting, you did what I did originally but I thought it wasn't worth it because only a few fields changed. Makes more sense though. I'll wait for github to tell me that it's not mergeable then. Thanks for the head up!

**Comment by GuillaumeGomez** [comment]
CI is happy.

**Review by JohnTitor — APPROVED** [review]
Thanks!

File: src/unix/bsd/freebsdlike/freebsd/freebsd11/mod.rs [hunk 1]
```diff
@@ -31,99 +31,182 @@ s! {
...
         pub ki_groups: [::gid_t; ::KI_NGROUPS],
+        /// Virtual size.
         pub ki_size: ::vm_size_t,
+        /// Current resident set size in pages.
         pub ki_rssize: ::segsz_t,
+        /// Resident set size before last swap.
         pub ki_swrss: ::segsz_t,
+        /// Text size (pages) XXX.
         pub ki_tsize: ::segsz_t,
+        /// Data size (pages) XXX.
```
Comment (Minoru) [inline src/unix/bsd/freebsdlike/freebsd/freebsd11/mod.rs:110]: What do these "XXX" markers mean here? Noticed them while doing a review for cargo-crev, got curious :) They're also present in all other versions of this struct.

Comment (GuillaumeGomez) [reply src/unix/bsd/freebsdlike/freebsd/freebsd11/mod.rs:110]: I have absolutely no idea. I tool the comment as is.

File: src/unix/bsd/freebsdlike/freebsd/freebsd11/mod.rs [hunk 2]
```diff
@@ -31,99 +31,182 @@ s! {
...
         pub ki_flag2: ::c_int,
+        /// Default FIB number.
         pub ki_fibnum: ::c_int,
+        /// Credential flags.
         pub ki_cr_flags: ::u_int,
+        /// Process jail ID.
         pub ki_jid: ::c_int,
+        /// Number of threads in total.
         pub ki_numthreads: ::c_int,
+        // Thread ID.
```
Comment (Minoru) [inline src/unix/bsd/freebsdlike/freebsd/freebsd11/mod.rs:188]: Shouldn't this be a doc comment? (Ditto for other versions of this struct)

Comment (GuillaumeGomez) [reply src/unix/bsd/freebsdlike/freebsd/freebsd11/mod.rs:188]: Yes it should. Wanna send a PR?

Comment (Minoru) [reply src/unix/bsd/freebsdlike/freebsd/freebsd11/mod.rs:188]: Sure, here it is: https://github.com/rust-lang/libc/pull/2563

Comment (GuillaumeGomez) [reply src/unix/bsd/freebsdlike/freebsd/freebsd11/mod.rs:188]: Thanks!

**Pull request merged**